decoder/src/frame.rs

```diff
@@ -91,6 +91,46 @@ impl<'t> Frame<'t> {
         DisplayMessage { frame: self }
     }
 
+    /// Returns an iterator over the fragments of the message contained in this log frame.
+    ///
+    /// Collecting this into a String will yield the same result as [`Self::display_message`], but
+    /// this iterator will yield interpolated fragments on their own. For example, the log:
+    ///
+    /// ```ignore
+    /// defmt::info!("foo = {}, bar = {}", 1, 2);
+    /// ```
+    ///
+    /// Will yield the following strings:
+    ///
+    /// ```ignore
+    /// vec!["foo = ", "1", ", bar = ", "2"]
+    /// ```
+    ///
+    /// Note that nested fragments will not yield separately:
+    ///
+    /// ```ignore
+    /// defmt::info!("foo = {}", Foo { bar: 1 });
+    /// ```
+    ///
+    /// Will yield:
+    ///
+    /// ```ignore
+    /// vec!["foo = ", "Foo { bar: 1 }"]
+    /// ```
+    ///
+    /// This iterator yields the same fragments as [`Self::fragments`], so you can zip them
+    /// together to get both representations.
+    pub fn display_fragments(&'t self) -> DisplayFragments<'t> {
+        DisplayFragments {
+            frame: self,
+            iter: self.fragments().into_iter(),
+        }
+    }
+
+    pub fn fragments(&'t self) -> Vec<Fragment<'t>> {
```

Comment:
Can we add documentation for this function?

Reply:
Done